cloudca/resource_cloudca_volume.go

```diff
@@ -0,0 +1,204 @@
+package cloudca
+
+import (
+	"fmt"
+	"github.com/cloud-ca/go-cloudca"
+	"github.com/cloud-ca/go-cloudca/api"
+	"github.com/cloud-ca/go-cloudca/services/cloudca"
+	"github.com/hashicorp/terraform/helper/schema"
+	"strings"
+)
+
+func resourceCloudcaVolume() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudcaVolumeCreate,
+		Read:   resourceCloudcaVolumeRead,
+		Update: resourceCloudcaVolumeUpdate,
+		Delete: resourceCloudcaVolumeDelete,
+
+		Schema: map[string]*schema.Schema{
+			"service_code": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "A cloudca service code",
+			},
+			"environment_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of environment where port forwarding rule should be created",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the volume to be created",
+			},
+			"zone_name": &schema.Schema{
```

Comment:
I thought that this could be dealt with the same way we deal with `service_code` and `environment_name` since its not something that the user can add themselves.